### DISCUSSION TITLE
Unexpected warning of Next.js TypeScript plugin and passing server action to client component

### DISCUSSION THREAD

**Discussion by ellemedit** [vercel/next.js#63656] [App Router]
I'm using next.js typescript plugin and got message:  `Props must be serializable for components in the "use client" entry file, "onSuccess" is invalid.ts(71007)`

```tsx
"use client";

export function MyButton({
  onSuccess // Props must be serializable for components in the "use client" entry file, "onSuccess" is invalid.ts(71007)
}: { onSuccess: () => Promise<void> }) {
  ...
}
```

https://nextjs.org/docs/app/building-your-application/data-fetching/server-actions-and-mutations#client-components

the docs say "You can also pass a Server Action to a Client Component as a prop" and it actually has no problem.

But the warning message keeps me focusing on it. Can we fix it?

**Comment by kyrregjerstad** [comment]
> I'm using next.js typescript plugin and got message: `Props must be serializable for components in the "use client" entry file, "onSuccess" is invalid.ts(71007)`
> 
> ```tsx
> "use client";
> 
> export function MyButton({
>   onSuccess // Props must be serializable for components in the "use client" entry file, "onSuccess" is invalid.ts(71007)
> }: { onSuccess: () => Promise<void> }) {
>   ...
> }
> ```
> 
> https://nextjs.org/docs/app/building-your-application/data-fetching/server-actions-and-mutations#client-components
> 
> the docs say "You can also pass a Server Action to a Client Component as a prop" and it actually has no problem.
> 
> But the warning message keeps me focusing on it. Can we fix it?

I'm facing the same issue, it only happens when `Select TypeScript Version` is set to `Use workspace version` in VS Code. Switching to VS Code's typescript version hides the message but at the cost of losing out on the extra NextJS type hints...
